ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -127,5 +137,67 @@ RnsShell::LayerInvalidateMask RSkComponentImage::updateComponentProps(const Shad
     }
     return updateMask;
 }
+
+sk_sp<SkImage> RSkComponentImage::getImageData(const char *path) {
+  sk_sp<SkImage> imageData{nullptr};
+  sk_sp<SkData> data;
+  if(!path) {
+    RNS_LOG_ERROR("Invalid File");
+    return nullptr;
+  }
+  data = SkData::MakeFromFileName(path);
+   if (!data) {
+    RNS_LOG_ERROR("Unable to make SkData for path : " << path);
+    return nullptr;
+  }
+  imageData = SkImage::MakeFromEncoded(data);
+  return imageData;
+}
+
+void RSkComponentImage::drawAndSubmit() {
+  layer()->client().notifyFlushBegin();
+  layer()->invalidate( RnsShell::LayerPaintInvalidate);
+  if (layer()->type() == RnsShell::LAYER_TYPE_PICTURE) {
+    RNS_PROFILE_API_OFF(getComponentData().componentName << " getPicture :", static_cast<RnsShell::PictureLayer*>(layer().get())->setPicture(getPicture()));
+  }
+  layer()->client().notifyFlushRequired();
+}
+
+void RSkComponentImage::getUriImage(ImageSource source) {
+  struct RemoteImageData* remoteImageData(new struct RemoteImageData);
+  auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  CurlRequest *curlRequest = new CurlRequest(nullptr,source.uri.c_str(),0,"GET");
+
+  // callback for remoteImageData
+  remoteImageData->callback = [&](const char* path, char* response, int size) {
+    // Responce callback from network. Get image data, insert in Cache and call Onpaint
+    sk_sp<SkData> data = SkData::MakeWithCopy(response,size);
+    sk_sp<SkImage> imageData = SkImage::MakeFromEncoded(data);
+    if(imageData) {
+    //Add in cache if image data is valid
+      imageCacheManagerInstance_.imageDataInsertInCche(path, imageData);
```

Comment:
need to check again here , if the data is present is cache for the uri.
if not available in cache, then need to add .

Reply:
If there are 5 networks requests for the same uri, how many entries are present in cache for the uri ?

Reply:
only one entry is present in the cache.